pkg/accounting/accounting.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package accounting
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/p2p"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+var _ Interface = (*Accounting)(nil)
+
+// Interface is the main interface for Accounting
+type Interface interface {
+	// Reserve reserves a portion of the balance for peer
+	// It returns an error if the operation would risk exceeding the disconnect threshold
+	// This should be called (always in combination with release) before a Credit action
+	Reserve(peer swarm.Address, price uint64) error
+	// Release releases reserved funds
+	Release(peer swarm.Address, price uint64)
+	// Credit increases the balance the peer has with us (we "pay" the peer)
+	Credit(peer swarm.Address, price uint64) error
+	// Debit increases the balance we have with the peer (we get "paid")
```

Comment:
We don't really pay in the Credit case either (hence "pay" instead of pay). We `plan to pay`. I would be careful to use the term pay anywhere in `Accounting`, payment happens exclusively in `Settlement`.

Reply:
Actually, we do a real payment when we do a credit, but we expect to get paid when we do a debit. Let me explain:

In terms of a credit, the payment consists of increasing the allowance for the other peer. 

In terms of debit, we just hope that the other peer did this for us (but we are not sure).

Reply:
But maybe this reasoning is too theoretical and not really worth diving into. In terms of the comments: if you prepend the comment between the brackets with `heuristic:`, then I think we are good to go.